sdk/lib/drivers/sound/mmixer/wave.c

```diff
@@ -244,13 +245,17 @@ MMixerOpenWavePin(
     LPMIXER_DATA MixerData;
     NTSTATUS Status;
     MIXER_STATUS MixerStatus;
+    DWORD cbSize;
 
     MixerData = MMixerGetDataByDeviceId(MixerList, DeviceId);
     if (!MixerData)
         return MM_STATUS_INVALID_PARAMETER;
 
+    /* Enforce 0 for WAVE_FORMAT_PCM, which ignores extra information size */
+    cbSize = WaveFormatEx->wFormatTag == WAVE_FORMAT_PCM ? 0 : WaveFormatEx->cbSize;
```

Comment:
It may be more consistent to check for `WAVE_FORMAT_EXTENSIBLE`, in order to match the test done in `MMixerInitializeDataFormat()` that copies the extra waveformatex stuff in that case.

Reply:
No, they are correct as is: they handle 2 different structures/cases. (See "my" discussion.)

Reply:
Yep, by using 'else' case, we're handling here other audio formats besides PCM and EXTENSIBLE too.

Reply:
So let me ask yet another last question:
in `MMixerInitializeDataFormat()`, is the:
```c
    /* Write additional fields for Extensible audio format */
    if (WaveFormatEx->wFormatTag == WAVE_FORMAT_EXTENSIBLE)
```
correct?
I suppose that the size of the `WAVEFORMATEXTENSIBLE` structure, obtained via:
```c
PWAVEFORMATEXTENSIBLE WaveFormatExt = (PWAVEFORMATEXTENSIBLE)&DataFormat->WaveFormatEx;
```
i.e. "hidden" in `DataFormat->WaveFormatEx`, was accounted for via the `cbSize` parameter given to the function?

Reply:
> correct?

Yes, as already replied.

> accounted for via the `cbSize` parameter

Yes, it is included in this struct member value.